MonstieWash/Assets/Scripts/MonsterAi/MoodArea.cs

```diff
@@ -57,6 +69,20 @@ private void Update()
         if (diminishingEffectiveness) currentEffectiveness = Mathf.Clamp(currentEffectiveness +( (diminishStrength/2f) * Time.deltaTime), 0f , 100f);
         //Reduce Cooldown over time.
         currentCooldown = Mathf.Clamp(currentCooldown -= Time.deltaTime, 0f, areaCooldown);
+
+        //Update shader based on recent touch
+        if (spriteToOutline.material.HasProperty("_TimeActive"))
+        {
+            var timeProperty = spriteToOutline.material.GetFloat("_TimeActive");
+
+            //Reset shader to original after timer runs out.
+            if (timeProperty <= 0) spriteToOutline.material = m_OriginalMat;
+
+            //Reduce Shaders TimeActive Property
+            spriteToOutline.material.SetFloat("_TimeActive", Mathf.MoveTowards(timeProperty,0,Time.deltaTime));
+        };
```

Comment:
This section needs to also check for whether the `spriteToOutline` is null to prevent errors when nothing is set.